### DISCUSSION TITLE
Property 'id' does not exist on type 'Session' using useSession with aumented type, any ideas?

### DISCUSSION THREAD

**Discussion by LFCavalcanti** [nextauthjs/next-auth#7854] [Help]
Hi,

So, I augmented the Session type as per documentation() to add the User ID provided with the JWT so it makes faster lookups for information with my backend.

Note: I'm using the AppRouter with Next 13.4.4 and Next-Auth 4.22.1

The type file is like this:
_./src/types/next-auth.d.ts_

`import NextAuth, { DefaultSession } from 'next-auth';
declare module 'next-auth' {
  interface Session  {
    user: {
      id?: string | undefined;
    } & DefaultSession['user'];
  }
}`

If I call the useSession hook on a client component, like this:
`const { data: session, status } = useSession();`

Then use the "id":
`body: JSON.stringify({ id: session?.id, forceResend }),`

Typescript raises the error: _Property 'id' does not exist on type 'Session'_

It actually is working, the property has the value since the Session callback is setup on the API, but the error stays there.

I don't know if this is an issue and reported here: https://github.com/nextauthjs/next-auth/issues/7132

My Session callback is like this:
`async session({ session, token }) {
      return {
        ...session,
        user: {
          ...session.user,
          id: token.sub,
        },
      };
    },`

**Comment by abdtriedcoding** [comment]
Facing same issue any updates ?

**Comment by adxthya** [comment]
Refer this:
https://next-auth.js.org/getting-started/typescript

**Comment by DigitalCreationsCo** [comment]
@LFCavalcanti I had a similar issue. I resolved it by place the d.ts files in a folder named 'types' in root redirectory. Not sure it worked. Try it out. Cheers.
